packages/ajna-contracts/contracts/ajna/ajna-actions/AjnaProxyActions.sol

```diff
@@ -469,7 +471,9 @@ contract AjnaProxyActions {
     uint256 oldPrice,
     uint256 newPrice
   ) public payable {
+    _validateBucketState(pool, convertPriceToIndex(newPrice));
     _supplyQuote(pool, amountToAdd, newPrice);
+    _validateBucketState(pool, convertPriceToIndex(newPrice));
```

Comment:
I would cache `convertPriceToIndex(newPrice)` in a variable since you are calling twice